c2corg_ui/views/document.py

```diff
@@ -432,12 +441,17 @@ def _redirect_to_full_url(self):
             raise HTTPInternalServerError(
                 "An error occurred while loading the document")
 
-        locale = data['locales'][0]
-        title = ''
-        if self._API_ROUTE == 'routes' and locale['title_prefix']:
-            title += locale['title_prefix'] + ' '
-        title += locale['title']
-        location = self.request.route_url(
-            self._API_ROUTE + '_view', id=data['document_id'],
-            lang=locale['lang'], slug=slugify(title))
-        raise HTTPMovedPermanently(location=location)
+        if 'redirects_to' in data:
+            if lang not in data['available_langs']:
+                lang = self._get_preferred_lang()
```

Comment:
When the API returns ``redirects_to`` it also provides the available langs. So it's easy to figure the best lang out as in your link.

But what should we do when the ``lang`` param is not provided (URL such as ``waypoints/123456``)? We don't know what langs are available for this document. Is there a way to ask the API for the available langs?